packages/backend/src/tenants/settings/service.test.ts

```diff
@@ -0,0 +1,277 @@
+import { Knex } from 'knex'
+import { Tenant } from '../model'
+import { TenantService } from '../service'
+import { faker } from '@faker-js/faker'
+import { getPageTests } from '../../shared/baseModel.test'
+import { Pagination, SortOrder } from '../../shared/baseModel'
+import { createTenantSettings, exchangeRatesSetting, randomSetting } from '../../tests/tenantSettings'
+import { TenantSetting } from './model'
+import {
+  CreateOptions,
+  GetOptions,
+  TenantSettingService,
+  UpdateOptions
+} from './service'
+import { AuthServiceClient } from '../../auth-service-client/client'
+
+describe('TenantSetting Service', (): void => {
+  let knex: Knex
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let tenant: Tenant
+  let tenantService: TenantService
+  let tenantSettingService: TenantSettingService
+  let authServiceClient: AuthServiceClient
+
+  const dbSchema = 'tenant_settings_service_test_schema'
+
+  beforeAll(async (): Promise<void> => {
+    deps = initIocContainer({ ...Config, dbSchema })
+    appContainer = await createTestApp(deps)
+
+    knex = await deps.use('knex')
+    tenantService = await deps.use('tenantService')
+    tenantSettingService = await deps.use('tenantSettingService')
+    authServiceClient = await deps.use('authServiceClient')
+  })
+
+  beforeEach(async (): Promise<void> => {
+    jest
+      .spyOn(authServiceClient.tenant, 'create')
+      .mockResolvedValueOnce(undefined)
+
+    tenant = await tenantService.create({
+      apiSecret: faker.string.uuid(),
+      email: faker.internet.email(),
+      idpConsentUrl: faker.internet.url(),
+      idpSecret: faker.string.uuid()
+    })
+  })
+
+  afterEach(async (): Promise<void> => {
+    await truncateTables(knex, true, dbSchema)
+  })
+
+  afterAll(async (): Promise<void> => {
+    nock.cleanAll()
+    await appContainer.shutdown()
+  })
+
+  describe('create', () => {
+    test('can create a tenant setting', async (): Promise<void> => {
+      const createOptions: CreateOptions = {
+        tenantId: tenant.id,
+        setting: [exchangeRatesSetting()]
+      }
+
+      const tenantSetting = await tenantSettingService.create(createOptions)
+
+      expect(tenantSetting).toEqual([
+        expect.objectContaining({
+          tenantId: tenant.id,
+          key: createOptions.setting[0].key,
+          value: createOptions.setting[0].value
+        })
+      ])
+    })
+
+    test('returns empty array if setting key is not allowed', async (): Promise<void> => {
+      const createOptions: CreateOptions = {
+        tenantId: tenant.id,
+        setting: [randomSetting()]
+      }
+
+      const tenantSetting = await tenantSettingService.create(createOptions)
+
+      expect(tenantSetting).toEqual([])
+    })
+  })
+
+  describe('get', () => {
+    let tenantSetting: TenantSetting[]
+
+    function createTenantSetting() {
+      const options: CreateOptions = {
+        tenantId: tenant.id,
+        setting: [randomSetting()]
```

Comment:
Double-checking here, since we are checking for valid keys in the create, would this end up inserting anything?